aws/resource_aws_iam_server_certificate_test.go

```diff
@@ -141,14 +145,27 @@ func testAccCheckCertExists(n string, cert *iam.ServerCertificate) resource.Test
 	}
 }
 
-func testAccCheckAWSServerCertAttributes(cert *iam.ServerCertificate) resource.TestCheckFunc {
+func getCertBody(body *string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		for _, rs := range s.RootModule().Resources {
+			if rs.Type != "tls_self_signed_cert" {
+				continue
+			}
+
+			*body = rs.Primary.Attributes["cert_pem"]
+		}
+		return nil
+	}
+}
+
+func testAccCheckAWSServerCertAttributes(cert *iam.ServerCertificate, certBody *string) resource.TestCheckFunc {
 	return func(s *terraform.State) error {
 		if !strings.Contains(*cert.ServerCertificateMetadata.ServerCertificateName, "terraform-test-cert") {
 			return fmt.Errorf("Bad Server Cert Name: %s", *cert.ServerCertificateMetadata.ServerCertificateName)
 		}
 
-		if *cert.CertificateBody != strings.TrimSpace(certBody) {
-			return fmt.Errorf("Bad Server Cert body\n\t expected: %s\n\tgot: %s\n", certBody, *cert.CertificateBody)
+		if *cert.CertificateBody != strings.TrimSpace(*certBody) {
+			return fmt.Errorf("Bad Server Cert body\n\t expected: %s\n\tgot: %s\n", *certBody, *cert.CertificateBody)
 		}
 		return nil
 	}
```

Comment:
The old certificate in `var certBody` on line 200 (apparently I cannot comment on lines outside of context) can go away too I think?